src/python/pants/backend/python/lint/flake8/subsystem.py

```diff
@@ -114,19 +131,26 @@ class Flake8LockfileSentinel(PythonToolLockfileSentinel):
 
 
 @rule(
-    desc="Determine all Python interpreter versions used by Flake8 in your project",
+    desc=(
+        "Determine all Python interpreter versions used by Flake8 in your project (for lockfile "
+        "usage)"
+    ),
     level=LogLevel.DEBUG,
 )
 async def setup_flake8_lockfile(
     _: Flake8LockfileSentinel, flake8: Flake8, python_setup: PythonSetup
 ) -> PythonLockfileRequest:
+    # While Flake8 will run in partitions, we need a single lockfile that works with every
+    # partition.
+    #
+    # This ORs all unique interpreter constraints. When paired with
+    # `InterpreterConstraints.partition_by_major_minor_versions`, the net effect is that every
+    # possible Python interpreter used will be covered.
```

Comment:
I wanted to dedupe, but we can't because of the `skip_field` options. I think the correctness could be important here. Let's say you don't run Pylint on your Py2 only code: we need to ignore those targets when determining the ICs for the lockfile.

Maybee we decide to give up on that correctness, but I want to start with a baseline of what is fully correct.

Reply:
Sure... but that filtering feels like it could be generalized?